dlgr/griduniverse/experiment.py

```diff
@@ -1720,8 +1720,117 @@ def analyze(self, data):
+        """
+        df = data.infos.df
+        dataState = df.loc[df['type'] == 'state']
+        if dataState.empty:
+            return {}
+        dlist = data.infos.list
+        all_events = [x for x in dlist if x[10] == 'event']
+
+        # get the unique origin_id for each player to differentiate players
+        moves = [x for x in all_events if 'move' in x[9]]
+        origin_ids = [set(x[11] for x in moves)][0]
+        origin_ids = list(origin_ids)
+
+        round_breaks = [x for x in all_events if "new_round" in x[9]] # find all the round dividers/breaks
+        num_rounds = len(round_breaks) + 1
+        if num_rounds == 1:
+            return {"round_number": 1, "round_data" : self.number_of_actions_per_round(origin_ids, moves)}
+        else:
+            number_of_actions_data = []
+            moves_and_round_breaks = [x for x in all_events if 'move' in x[9] or 'new_round' in x[9]]
+            # split the move data of entire game into lists containing  move data for each round
+            rounds_moves = self.ssplit2(moves_and_round_breaks, tuple(round_breaks))
+            # parse each rounds moves, one at a time
+            round_number = 1
+            for moves in rounds_moves:
+                # note that the round number might not match how griduniverse numbers the rounds
+                # however the system used here to number rounds is verified to be accurate chronologically
+                # round 1 happened before round 2 etc
+                data_dict = {"round_number": round_number,
+                             "round_data" : self.number_of_actions_per_round(origin_ids, moves) }
+                number_of_actions_data.append(data_dict)
+                round_number += 1
+
+        return number_of_actions_data
+
+    def average_time_to_start(self, data):
+        """ the average time to start the game.
+            compare the time of participant's first move info to the network creation time
+        """
+        df = data.infos.df
+        dataState = df.loc[df['type'] == 'state']
+        if dataState.empty:
+            return str(datetime.timedelta(0))
+        network_creation_time = data.networks.list[0][1]
+        dlist = data.infos.list
+        all_events = [x for x in dlist if x[10] == 'event']
+        moves = [x for x in all_events if 'move' in x[9]]
+
+        # get the unique origin_id for each player to differentiate players
+        origin_ids = [set(x[11] for x in moves)][0]
+        origin_ids = list(origin_ids)
+
+        deltasum = datetime.timedelta(0) # init
+        deltas = []
+        for player in origin_ids:
+            # get all players moves
+            players_moves = [x for x in moves if x[11] == player]
+            if len(players_moves) != 0:  # is it possible that the player does not move?
+                # use the time of their first move
+                delta = players_moves[0][1] - network_creation_time
+                deltas.append(delta)
```

Comment:
If all you're using deltas for is for calculating the mean you can keep a count rather an a full list, which may be very long.